pkg/metadata/metadatapb/rpc.proto

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+syntax = "proto3";
+package thanos;
+
+import "store/storepb/types.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "metadatapb";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+
+// Do not generate XXX fields to reduce memory footprint and opening a door
+// for zero-copy casts to/from prometheus data types.
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+service Metadata {
+  rpc Metadata(MetadataRequest) returns (stream MetadataResponse);
+}
+
+message MetadataRequest {
+  string metric = 1;
+  int32 limit = 2;
+  PartialResponseStrategy partial_response_strategy = 3;
+}
+
+message MetadataResponse {
+  oneof result {
+    /// group for rule groups. It is up to server implementation to decide how many of those to put here within single frame.
```

Comment:
Comment should probably start with metadata? something cut off?

Reply:
I missed this. Thanks!